### ISSUE TITLE
[BUG] certain ticks omitted if x axis is a categorial string

### ISSUE THREAD

**Issue by oheil** [JuliaPlots/Plots.jl#3063]
scatter (and plots) omit certain ticks if the x axis consists of categorial string values.

MWE:
```
using Plots
using DataFrames
data=DataFrame(x=string.(1:23),y=rand(Float64,23))
scatter(data[:x],data[:y],xaxis="x",yaxis="y")
```

![omitted_ticks](https://user-images.githubusercontent.com/5408557/95982798-de860700-0e20-11eb-8d34-70f66dead56b.jpg)

### Backends

This bug occurs on ( insert `x` below )

```
Backend      | yes | no  | untested
-------------|-----|-----|---------
gr (default) |  x  |     |
pyplot       |  x  |     |
plotly       |  x  |     |
plotlyjs     |  x  |     |
pgfplotsx    |     |     | x
inspectdr    |     |     | x
```

### Versions

Plots.jl version:
`Plots v0.28.4`
Backend  version (`]st -m`):
```
PyPlot v2.9.
PlotlyJS v0.14.0
```

Output of `versioninfo()`:
```
Julia Version 1.5.0
Commit 96786e22cc (2020-08-01 23:44 UTC)
Platform Info:
  OS: Windows (x86_64-w64-mingw32)
  CPU: Intel(R) Core(TM) i7-4770 CPU @ 3.40GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-9.0.1 (ORCJIT, haswell)
Environment:
  JULIA_PKGDIR = c:\Program Files\Julia-0.6.2\packages\
  JULIA_SHELL = C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe
```

**Comment by daschw** [comment]
You can add `xticks = :all` to your `scatter` command.

**Comment by oheil** [comment]
Still it is surprising, that some ticks are omitted. That there exists a workaround doesn't mean it isn't a bug. Of course this is up to you to decide.

**Comment by daschw** [comment]
I agree that defaulting to `ticks = :all` in these cases would be a better default.

**Comment by daschw** [comment]
However, defaults are always tricky. Consider
```julia
x = [string(letter, num) for letter in 'a':'z' for num in 0:9]
```
and compare
```julia
plot(x, rand(length(x)))
```
![ticks_auto](https://user-images.githubusercontent.com/16589944/97233210-d8a80280-17de-11eb-8f2a-4079e856ff3b.png)
to
```julia
plot(x, rand(length(x)), ticks = :all)
```
![ticks_all](https://user-images.githubusercontent.com/16589944/97233230-e5c4f180-17de-11eb-8507-3ce3f95042d7.png)

**Comment by oheil** [comment]
Omitting ticks because of lack of space is reasonable but only in a regular way like "omit every second tick" or "only plot every fifth tick". Omitting ticks, as in my starting example of this issue, where ticks are missing without any obvious reason in a irregular way, as if it is "lets check the next tick: plot it yes or no?" doesn't make too much sense and still I consider this behaviour as a bug (remark: i didn't check the code).
So it seems, that the question "ticks = :all" is default, yes or no, is misleading, despite using it would be a workaround for my specific example.
